tofu/geom/_core.py

```diff
@@ -6180,19 +6180,9 @@ def calc_signal(
             #    and interferometer)
             val = func(pts, t=t, vect=vect)
             # Integrate
-            if val.ndim == 2:
-                sig = np.full((val.shape[0], self.nRays), np.nan)
-            else:
-                sig = np.full((1, self.nRays), np.nan)
-            indpts = np.r_[0, indpts, pts.shape[1]]
-            for ii in range(0, self.nRays):
-                sig[:, ii] = (
-                    np.nansum(
-                        val[:, indpts[ii]:indpts[ii + 1]],
-                        axis=-1
-                    )
-                    * reseff[ii]
-                )
+            sig = np.add.reduceat(val, np.r_[0, indpts],
```

Comment:
More concise and numerically faster, only advantages :-)
Usable in _GG.LOS_calc_signal(method='sum', minimize='calls') ?
Usable elsewhere ?

Compatible with parallelization ?

Reply:
nope, only functions that don't require gil can be parallelized, calling numpy functions requires the gil, so it is not possible. This is why it wasn't parallelized with newcalc=True